cognite/client/data_classes/_base.py

```diff
@@ -281,14 +280,31 @@ def get(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Op
             return self._id_to_item.get(id)
         return self._external_id_to_item.get(external_id)
 
-    def to_pandas(self, camel_case: bool = False) -> pandas.DataFrame:
+    def to_pandas(
+        self,
+        camel_case: bool = False,
+        expand_metadata: bool = False,
+        metadata_prefix: str = "metadata.",
+    ) -> pandas.DataFrame:
         """Convert the instance into a pandas DataFrame.
 
+        Args:
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata column into separate columns.
+            metadata_prefix (str): Prefix to use for metadata columns.
+
         Returns:
             pandas.DataFrame: The dataframe.
         """
         pd = cast(Any, utils._auxiliary.local_import("pandas"))
         df = pd.DataFrame(self.dump(camel_case=camel_case))
+
+        if expand_metadata and "metadata" in df.columns:
+            meta_df = pd.json_normalize(df["metadata"]).add_prefix(metadata_prefix)
+            if common_cols := set(meta_df.columns).intersection(set(df.columns)):
+                raise ValueError(f"Metadata contains columns that are already present in the dataframe: {common_cols}")
+            df = pd.concat([df.drop("metadata", axis=1), meta_df], axis=1)
```

Comment:
Question: Using `join` already gives a quite clear error message when columns overlap, example:
```py
pd.DataFrame({"id": [1,2,3], "foo": list("asd")}).join(
    pd.DataFrame({"foo": list("xyz")})
)
# --> Raises ValueError ...
#     ...columns overlap but no suffix specified: Index(['foo'], dtype='object')
```
Maybe we don't have to write our own error handling? 🤔

Reply:
Good point! With the concatenate function, this can be achieved using the `verify_integrity` flag. See attached. What are your opinions @haakonvt and @doctrino? I am in favour of taking away the freedom from the user here and forcing the verify integrity, and then removing the error checking to simplify the codebase/addition.

<img width="1172" alt="Screenshot 2023-07-25 at 12 28 18" src="https://github.com/cognitedata/cognite-sdk-python/assets/60771923/ae0681e8-835e-47ee-a4ec-e6e7cbcc0cd5">
Figure 1: Without verify_integrity flag.

<img width="1172" alt="Screenshot 2023-07-25 at 12 29 39" src="https://github.com/cognitedata/cognite-sdk-python/assets/60771923/546d7ba2-61a4-4b47-a671-6bd8d4313110">
Figure 2: With verify_integrity flag.

Reply:
Nevermind, I read these comments in reverse order. 😅 I see now that the concat function would be replaced by the join by using the speedier implementation.